frontend/app/select-template/page.tsx

```diff
@@ -1,18 +1,67 @@
 "use client"
 
 import { useState } from "react"
+import Link from "next/link"
 import { IPageType, ITemplate } from "@/api/generated"
+import { FilePlus2 } from "lucide-react"
 
 import { Button } from "@/components/ui/button"
 import Card from "@/components/SelectTemplate/Card"
 import SelectTemplateSideBar from "@/components/SelectTemplate/Sidebar/Sidebar"
-import { FilePlus2 } from "lucide-react"
-import Link from "next/link"
 
 export default function IndexPage() {
   const [template, setTemplates] = useState<ITemplate[]>([])
   const [pageType, setPageType] = useState<IPageType>(IPageType.NONE)
 
+  function getTemplateId(template: ITemplate): string {
```

Comment:
16-64, This code can most likely be split up into a helper function, so it's easier to refactor and not as tightly coupled within the application. getSortedTemplatesByFavorite makes sense to stay here though if you want. unless you make a functional map component.

I would move this to /frontend/utils

Reply:
Also for me to test later: What happens if the key isn't valid

Reply:
Good idea. I will move it there later this week thanks

Reply:
Hey just following up on this PR, branches have diverged quite a bit.

I'm happy to do the cleanup here if you don't have the time, or would you have time available to do the above changes?